GherkinSyncTool.Synchronizers.AllureTestOps/Client/AllureClientWrapper.cs

```diff
@@ -62,42 +62,187 @@ public TestCaseOverview GetTestCaseOverview(ulong id)
             return response.Content;
         }
 
-        public void UpdateTestCase(TestCaseContent currentCase, TestCaseRequest caseToUpdate)
+        public void UpdateTestCase(TestCaseContent currentCase, CreateTestCaseRequestExtended caseToUpdate)
         {
-            if (!IsTestCaseContentEqual(currentCase, caseToUpdate))
-            {
-                var response = _allureClient.UpdateTestCaseAsync(currentCase.Id, caseToUpdate).Result;
+            var testCaseOverview = _allureClient.GetTestCaseOverviewAsync(currentCase.Id).Result.Content;
+            var contentEqual = IsTestCaseContentEqual(testCaseOverview, caseToUpdate);
+            bool updated = false;
 
+            bool scenarioIsEqual;
+            //Remove scenario
+            if (testCaseOverview!.Scenario is not null && caseToUpdate.CreateTestCaseRequest.Scenario is null)
+            {
+                var response = _allureClient.DeleteTestCaseScenarioAsync(testCaseOverview.Id).Result;
                 ValidateResponse(response);
 
-                Log.Info($"Updated: [{currentCase.Id}] {caseToUpdate.Name}");
+                scenarioIsEqual = true;
+                updated = true;
             }
             else
             {
-                Log.Info($"Up-to-date: [{currentCase.Id}] {currentCase.Name}");
+                scenarioIsEqual = IsScenarioEqual(testCaseOverview, caseToUpdate);
+            }
+
+            if (!contentEqual && scenarioIsEqual && caseToUpdate.CreateTestCaseRequest.Scenario is not null)
+            {
+                for (var i = 0; i < caseToUpdate.CreateTestCaseRequest.Scenario.Steps.Count; i++)
+                {
+                    var step = caseToUpdate.CreateTestCaseRequest.Scenario.Steps[i];
+                    step.Attachments = testCaseOverview.Scenario!.Steps[i].Attachments;
+                }
+            }
+
+            if (!contentEqual || !scenarioIsEqual)
+            {
+                var response = _allureClient.UpdateTestCaseAsync(currentCase.Id, caseToUpdate.CreateTestCaseRequest).Result;
+                ValidateResponse(response);
+                updated = true;
+            }
+
+            Log.Info(updated
+                ? $"Updated: [{currentCase.Id}] {caseToUpdate.CreateTestCaseRequest.Name}"
+                : $"Up-to-date: [{currentCase.Id}] {currentCase.Name}");
+        }
+
+        private bool IsScenarioEqual(TestCaseOverview testCaseOverview, CreateTestCaseRequestExtended caseToUpdate)
+        {
+            if (testCaseOverview.Scenario is null && caseToUpdate.CreateTestCaseRequest.Scenario is null)
+            {
+                return true;
+            }
+
+            if (testCaseOverview.Scenario is null && caseToUpdate.CreateTestCaseRequest.Scenario.Steps is not null)
+            {
+                if (caseToUpdate.StepsAttachments.Any())
+                {
+                    UpdateTestCaseStepAttachments(caseToUpdate, testCaseOverview);
```

Comment:
Not quite understand why equality comparer updates something

Reply:
Good point! Fixed.